forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,134 @@
+                example, "text" attribute of Token.
+    """
+    def __init__(self, config: Union[Dict, Config]):
+        super().__init__(config)
+
+        if "attribute" not in self.config:
+            raise AttributeError("attribute needs to be specified in "
+                                "the configuration of an AttributeExtractor.")
+
+    @property
+    def state(self) -> Dict:
+        state = super().state
+        state.update({
+            "attribute": self.config.attribute
+        })
+        return state
+
+    @classmethod
+    def from_state(cls, state: Dict) \
+        -> Union["BaseExtractor", "AttributeExtractor"]:
+        obj = BaseExtractor.from_state(state)
+        additional = {
+            "attribute": state["attribute"]
+        }
+        obj.config = Config(obj.config, additional,
+                            allow_new_hparam=True)
+        obj.__class__ = cls
+        return obj
+
+    @staticmethod
+    def get_attribute(entry: Entry, attr: str) -> Any:
+        r"""Functionality: Get the attribute from entry. You can
+        overwrite this function if you have sepcial way to get the
+        attribute from entry.
+        """
+        return getattr(entry, attr)
+
+    @staticmethod
+    def set_attribute(entry: Entry, attr: str, value: Any):
+        r"""Functionality: Set the attribute of an entry to value.
+        You can overwrite this function if you have sepcial way to
+        set the attribute.
+        """
+        if attr == "text":
+            raise AttributeError("text attribute of entry cannot "
+                                "be changed.")
+        setattr(entry, attr, value)
+
+    def update_vocab(self, pack: DataPack, instance: Annotation):
+        r"""Functionality: Get all attributes of one instance and
+        add them into the vocabulary.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            self.add(self.get_attribute(entry, self.config.attribute))
+
+    def extract(self, pack: DataPack, instance: Annotation) -> Feature:
+        r"""Functionality: Extractor the attributes of one instance.
+        For example, the text of tokens in one sentence.
+        """
+        data = []
+        for entry in pack.get(self.config.entry_type, instance):
+            value = self.get_attribute(entry, self.config.attribute)
+            rep = self.element2repr(value) if self.vocab else value
+            data.append(rep)
+
+        meta_data = {"pad_value": self.get_pad_value(),
+                     "dim": 1,
+                     "dtype": int if self.vocab else Any}
+
+        return Feature(data=data,
+                       metadata=meta_data,
+                       vocab=self.vocab)
+
+    def remove_from_pack(self, pack: DataPack, instance: Annotation):
+        r"""Funcationality: Remove attributes of one instance. For
+        example remove all pos tags of tokens in one sentence, if the
+        entry_type is Token and the attribute is pos.
```

Comment:
Please mention the purpose of this function (is to before evaluation)

Further, I would suggest renaming `remove_from_pack` to `pre_evaluation_action`

Reply:
Got it.